### PR TITLE
chore: ts support

### PR DESCRIPTION
**Pull request by anri-asaturov**
https://github.com/PeerioTechnologies/peerio-icebear/pull/257

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+module.exports = function(api) {
+    api.cache.never();
```
Comment (lostfictions) [inline babel.config.js]: why the change to .js and this line?

Comment (anri-asaturov) [reply babel.config.js]: The change - because it's the new and recommended way, and, more importantly it's gonna be the sole babel config used. It's gonna ignore any other configs in the tree.

This line - because I want to see how slow it really is first and because it has some options I don't have time to figure out until there's a need, feel free to change if you do.

Comment (anri-asaturov) [reply babel.config.js]: So the previous setup did merge babel configs from icebear and desktop. This one is safe.

Comment (lostfictions) [reply babel.config.js]: Still not sure I understand the line, sorry :(  It's slower but it saves us from having to figure something out right now?

Comment (anri-asaturov) [reply babel.config.js]: Ah, no,
Using babel.config.js disables the cascade. https://babeljs.io/docs/en/next/babelconfigjs.html

Calling `api.cache.never();` disables babel's internal cache. I'm trying to find where did I read about it and what made me disable it.

Comment (lostfictions) [reply babel.config.js]: The only mention I can find of it is this:
screenshot from 2018-07-25 16-57-31
https://babeljs.io/docs/en/next/babelconfigjs#apicache

Which worries me...

**Comment by lostfictions** [comment]
So is this supposed to run currently? I merged in `dev` and `npm start` throws before opening a window?

File: babel.config.js (hunk 1 above)
Comment (anri-asaturov) [reply babel.config.js]: No-no, I've seen the full docs somewhere, will find it soon.

**Comment by anri-asaturov** [comment]
@lostfictions eh. it did work yes, except eslint was spewing errors about undef vars.

**Comment by lostfictions** [comment]
Doesn't work for me, sorry :(
```
$ git checkout typescript
$ git pull
$ rm -rf node_modules && rm -rf app/node_modules
$ npm i
$ npm start
```

```
[electron:run  ] App threw an error during load
[electron:run  ] TypeError: Class extends value undefined is not a constructor or null
[electron:run  ]     at Object.<anonymous> (/home/six/Code/desktop/app/build/stores/indexed-db-storage.js:9:32)
[electron:run  ]     at Object.<anonymous> (/home/six/Code/desktop/app/build/stores/indexed-db-storage.js:149:3)
[electron:run  ]     at Module._compile (module.js:642:30)
[electron:run  ]     at Object.Module._extensions..js (module.js:653:10)
[electron:run  ]     at Module.load (module.js:561:32)
[electron:run  ]     at tryModuleLoad (module.js:504:12)
[electron:run  ]     at Function.Module._load (module.js:496:3)
[electron:run  ]     at Module.require (module.js:586:17)
[electron:run  ]     at require (internal/module.js:11:18)
[electron:run  ]     at Object.<anonymous> (/home/six/Code/desktop/app/build/config.js:17:21)
[electron:run  ]     at Object.<anonymous> (/home/six/Code/desktop/app/build/config.js:105:3)
[electron:run  ]     at Module._compile (module.js:642:30)
[electron:run  ]     at Object.Module._extensions..js (module.js:653:10)
[electron:run  ]     at Module.load (module.js:561:32)
[electron:run  ]     at tryModuleLoad (module.js:504:12)
[electron:run  ]     at Function.Module._load (module.js:496:3)
[electron:run  ] uncaughtException in Node: TypeError: Class extends value undefined is not a constructor or null
[electron:run  ]     at Object.<anonymous> (/home/six/Code/desktop/app/build/stores/indexed-db-storage.js:9:32)
[electron:run  ]     at Object.<anonymous> (/home/six/Code/desktop/app/build/stores/indexed-db-storage.js:149:3)
[electron:run  ]     at Module._compile (module.js:642:30)
[electron:run  ]     at Object.Module._extensions..js (module.js:653:10)
[electron:run  ]     at Module.load (module.js:561:32)
[electron:run  ]     at tryModuleLoad (module.js:504:12)
[electron:run  ]     at Function.Module._load (module.js:496:3)
[electron:run  ]     at Module.require (module.js:586:17)
[electron:run  ]     at require (internal/module.js:11:18)
[electron:run  ]     at Object.<anonymous> (/home/six/Code/desktop/app/build/config.js:17:21)
[electron:run  ]     at Object.<anonymous> (/home/six/Code/desktop/app/build/config.js:105:3)
[electron:run  ]     at Module._compile (module.js:642:30)
[electron:run  ]     at Object.Module._extensions..js (module.js:653:10)
[electron:run  ]     at Module.load (module.js:561:32)
[electron:run  ]     at tryModuleLoad (module.js:504:12)
[electron:run  ]     at Function.Module._load (module.js:496:3)
[electron:run  ] Chat list sorting.
```

File: babel.config.js (hunk 1 above)
Comment (anri-asaturov) [reply babel.config.js]: this is so weird, i remember reading an article or a doc page (just could not have imagined the api out of thin air), but all i can find now is https://github.com/babel/babel/blob/master/packages/babel-core/src/config/caching.js
